lib/Model/FileCacheWrapper.php

```diff
@@ -451,12 +451,19 @@ public function toCache(): array {
 	 * @return bool
 	 */
 	public function isAccessible(): bool {
-		if ($this->getId() === 0 || $this->getPath() === '') {
+		if ($this->getId() === 0) {
 			return false;
 		}
 
-		return !(explode('/', $this->getPath(), 2)[0] !== 'files'
-				 && explode(':', $this->getStorage(), 2)[0] === 'home');
+		$path = $this->getPath();
+		[$storageType,] = explode('::', $this->getStorage(), 2);
+
+		if ($path === '') {
+			// we only accept empty path on external storage
+			return (in_array($storageType, ['ftp', 'sftp', 'swift', 'smb', 'amazon']));
```

Comment:
Do we have this list as a dynamic array somewhere?
I am afraid we will forget to update this one if we support a new external storage

Reply:
@Altahrim added a notice in case of an eventual unsupported storage
@szaimen  I don't think you can share your full (top-root) home

Reply:
What I meant was that local storage is also a kind of external storage :)

Reply:
```suggestion
			return (in_array($storageType, ['ftp', 'sftp', 'swift', 'smb', 'amazon', 'local']));
```

Reply:
btw, arent there even further kinds?

Reply:
@szaimen ok, got it, fixing :D

Reply:
added local and webdav